src/ArmTemplates/Commands/Executors/CreatorExecutor.cs

```diff
@@ -205,7 +206,7 @@ public async Task ExecuteGenerationBasedOnConfiguration()
 
             this.logger.LogInformation("Creating tag template");
             this.logger.LogInformation("------------------------------------------");
-            var tagTemplate = this.creatorParameters.Tags != null ? this.tagTemplateCreator.CreateTagTemplate(this.creatorParameters) : null;
+            var tagTemplate = !this.creatorParameters.Tags.IsNullOrEmpty() || this.creatorParameters.Apis.Any(x => !x.Tags.IsNullOrEmpty()) ? this.tagTemplateCreator.CreateTagTemplate(this.creatorParameters) : null;
```

Comment:
Let's move all the section of `tagTemplate` creation to a separate method, as it is done it `ExtractorExecutor` - there in the `GenerateTemplates` method we call others methods, that encapsulate logic of creating that or another template.

I would go with this approach for several reasons:
- easy testing of some occasional use-cases - we do not need to run whole `ExecuteBasedOnConfiguration` only to check `tag-template`
- easy debug of failures - stack trace tells you directly what method failed. Currently you will have to find "that specific line" of failure in `ExecuteBasedOnConfiguration` method
- I know it is quite early to think about that, but some users requested `resource-kit` as a NuGet. So `ExtractorExecutor` has the API for that:
```
var extractorExecutor = new ExtractorExecutor(...);
await extractorExecutor.GenerateTagTemplates();
```
But in current implementation you do not have a possibility to do this. If it is not that hard to (and I think it is not), then it is better to already move logic to separate methods. Also don't forget to remove that dummy `-------` log :D